cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -90,90 +103,102 @@ renderShelleyQueryCmdError err =
+  eraInMode <- hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch anyEra (AnyConsensusMode consensusMode))
+                 $ toEraInMode era consensusMode
+
   let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
 
-  anyE@(AnyCardanoEra era) <- determineEra cModeParams localNodeConnInfo
-  let cMode = consensusModeOnly cModeParams
-  sbe <- getSbe $ cardanoEraStyle era
+  qInMode <- case cardanoEraStyle era of
+               LegacyByronEra -> left ShelleyQueryCmdByronEra
+               ShelleyBasedEra sbe -> return . QueryInEra eraInMode
+                                        $ QueryInShelleyBasedEra sbe QueryProtocolParameters
 
-  case toEraInMode era cMode of
-    Just eInMode -> do
-      let query = QueryInEra eInMode
-                    $ QueryInShelleyBasedEra sbe QueryProtocolParameters
-      result <- executeQuery
-                  era
-                  cModeParams
-                  localNodeConnInfo
-                  query
-      writeProtocolParameters mOutFile result
-    Nothing -> left . ShelleyQueryCmdEraConsensusModeMismatch anyE $ AnyConsensusMode cMode
- where
-  writeProtocolParameters
-    :: Maybe OutputFile
-    -> ProtocolParameters
-    -> ExceptT ShelleyQueryCmdError IO ()
-  writeProtocolParameters mOutFile' pparams =
-    case mOutFile' of
-      Nothing -> liftIO $ LBS.putStrLn (encodePretty pparams)
-      Just (OutputFile fpath) ->
-        handleIOExceptT (ShelleyQueryCmdWriteFileError . FileIOError fpath) $
-          LBS.writeFile fpath (encodePretty pparams)
+  res <- liftIO $ queryNodeLocalState localNodeConnInfo Nothing qInMode
+  case res of
+    Left acqFailure -> left $ ShelleyQueryCmdAcquireFailure acqFailure
+    Right ePparams ->
+      case ePparams of
+        Left err -> left . ShelleyQueryCmdLocalStateQueryError $ EraMismatchError err
+        Right pparams -> writeProtocolParameters mOutFile pparams
+
+writeProtocolParameters
+  :: Maybe OutputFile
+  -> ProtocolParameters
+  -> ExceptT ShelleyQueryCmdError IO ()
+writeProtocolParameters mOutFile pparams =
+  case mOutFile of
+    Nothing -> liftIO $ LBS.putStrLn (encodePretty pparams)
+    Just (OutputFile fpath) ->
+      handleIOExceptT (ShelleyQueryCmdWriteFileError . FileIOError fpath) $
+        LBS.writeFile fpath (encodePretty pparams)
 
 runQueryTip
-  :: AnyConsensusModeParams
+  :: AnyCardanoEra
+  -> AnyConsensusModeParams
   -> NetworkId
   -> Maybe OutputFile
   -> ExceptT ShelleyQueryCmdError IO ()
-runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
-  SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
-  let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
-      consensusMode = consensusModeOnly cModeParams
+runQueryTip (AnyCardanoEra era) (AnyConsensusModeParams cModeParams) network mOutFile = do
+    SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
```

Comment:
We should avoid semantically null changes, like indentation changes.

Reply:
Same as above. If you look at the diff we have clearly undone some refactoring.

Reply:
> 1. I would like to request a change to the pool-params output to make it more clear what the three different sets returned mean. See suggestion below.
> 2. In addition to this, I would like for the future set an additional field called `activeInEpoch` containing what epoch the future pool update becomes active.
> 
> **Current output:**
> An array containing three unnamed sets for current, future, and retired.
> 
> ```
> [
>   {
>     "publicKey": "0b08b536ba3afe21fbd826fb92bee897a3d1e575a7849b2e6c4659ab",
>     "...": "rest of the fields"
>   },
>   {
>     "publicKey": "0b08b536ba3afe21fbd826fb92bee897a3d1e575a7849b2e6c4659ab",
>     "...": "rest of the fields"
>   },
>   null
> ]
> ```
> 
> **Requested output:**
> My idea would be to have a schema like this instead, replacing the JSON array containing three unnamed objects with a key/value schema where `active` and `future` contains a JSON object(null for future if there are none) and `retired` the epoch number or null.
> Makes it clean and easy to understand what is what.
> 
> ```
> {
>   "active": {
>     "publicKey": "0b08b536ba3afe21fbd826fb92bee897a3d1e575a7849b2e6c4659ab",
>     "...": "rest of the fields"
>   },
>   "future": {
>     "publicKey": "0b08b536ba3afe21fbd826fb92bee897a3d1e575a7849b2e6c4659ab",
>     "activeInEpoch": 125,
>     "...": "rest of the fields"
>   },
>   "retired": null
> }
> ```

Added